sashimi-webapp/src/components/editor-viewer/Viewer.vue

```diff
@@ -19,37 +13,133 @@
 <script>
   import Vue from 'vue';
   import AsyncComputed from 'vue-async-computed';
+  import _ from 'lodash';
   import documentPackager from 'src/logic/documentPackager';
-
-  Vue.use(AsyncComputed);
+  import PageRenderer from 'src/logic/renderer';
   
+  Vue.use(AsyncComputed);
+
+  // Declare page renderers instance. These renderer
+  // will be initialised when this component is mounted.
+  const pageRendererList = {
+    slides: null,
+    pages: null,
+  };
+
+  // Throttle function used to limit the rate which
+  // the render function is called
+  const throttleTime = 600;
+  const renderThrottleFn = _.throttle((markdownString, pageRenderer) => {
+    documentPackager.getHtmlData(markdownString)
+    .then(htmlString => pageRenderer.write(htmlString));
+  }, throttleTime);
+
+  const updateViewer = ((compunent) => {
```

Comment:
Nothing significant, but i believe you meant to type 'component' instead of 'compunent' right? Haha